v2/cli/init_command.go

```diff
@@ -0,0 +1,255 @@
+		if overwrite, err = checkFileOverwrite(scriptPath); err != nil {
+			return err
+		} else if overwrite {
+			if err = ioutil.WriteFile(
+				scriptPath,
+				scriptBytes,
+				0644,
+			); err != nil {
+				return err
+			}
+		}
+	}
+
+	// Script is added to project.yaml otherwise
+	var projectBytes []byte
+	projectBytes, err = execTemplate(projectTemplate, fields)
+	if err != nil {
+		return err
+	}
+
+	// Check if the .brigade/project.yaml path exists.
+	// Confirm overwrite if so.
+	projectPath := path.Join(".brigade", "project.yaml")
+	if overwrite, err = checkFileOverwrite(projectPath); err != nil {
+		return err
+	} else if overwrite {
+		if err = ioutil.WriteFile(
+			projectPath,
+			projectBytes,
+			0644,
+		); err != nil {
+			return err
+		}
+	}
+
+	var notesBytes []byte
+	notesBytes, err = execTemplate(notesTemplate, fields)
+	if err != nil {
+		return err
+	}
+
+	// Create a NOTES.txt file
+	notesPath := path.Join(".brigade", "NOTES.txt")
+	if overwrite, err = checkFileOverwrite(notesPath); err != nil {
+		return err
+	} else if overwrite {
+		if err = ioutil.WriteFile(
+			notesPath,
+			notesBytes,
+			0644,
+		); err != nil {
+			return err
+		}
+	}
+
+	var secretsBytes []byte
+	secretsBytes, err = execTemplate(secretsTemplate, fields)
+	if err != nil {
+		return err
+	}
+
+	// Create a secrets.yaml file
+	secretsPath := path.Join(".brigade", "secrets.yaml")
+	if overwrite, err = checkFileOverwrite(secretsPath); err != nil {
+		return err
+	} else if overwrite {
+		if err = ioutil.WriteFile(
+			secretsPath,
+			secretsBytes,
+			0644,
+		); err != nil {
+			return err
+		}
+	}
+
+	err = addToGitignore(".brigade/secrets.yaml", "./.gitignore")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Check if the dirPath directory exists. If not, create it
+func checkCreateDirectory(dirPath string) error {
+	if dirExists, err := file.Exists(dirPath); err != nil {
+		return err
+	} else if !dirExists {
+		fmt.Println("Creating " + dirPath + " directory...")
+		err = os.Mkdir(dirPath, 0755)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Detect if .gitignore exists. If it does, add .brigade/secrets.yaml to it
+// If not, create it and add .brigade/secrets.yaml to it unconditionally
+func addToGitignore(pathToIgnore string, gitIgnorePath string) error {
+	file, err :=
+		os.OpenFile(gitIgnorePath, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	if _, err =
+		file.WriteString(fmt.Sprintf("\n%s\n", pathToIgnore)); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Given an id, check if it is valid
+func idMatch(id string) error {
+	regex := regexp.MustCompile(`^[a-z][a-z\d-]*[a-z\d]$`)
+	idMatch := regex.MatchString(id)
+	if !idMatch || len(id) < 3 || len(id) > 63 {
+		return fmt.Errorf("invalid value %q for --id flag"+
+			" (3-63 characters, first char must be"+
+			" a letter, lowercase letters only)", id)
+	}
+	return nil
+}
```

Comment:
This has, weirdly, been the first time we've cared about validating a project ID client side, but it's unlikely to be the last. Let's move this to a function named `ValidateProjectID` in the `sdk/v2/core` package.

Reply:
Would I be moving this into the projects.go file? That would make the most sense, but it seems like most functions in that file are related to a ProjectsClient

Reply:
`projects.go` sounds like a good place. That file is all things project-related and not limited to the API client.